controller/AuthoringTool.php

```diff
@@ -84,56 +85,71 @@ protected function getValidatedLtiMessagePayload(): LtiMessagePayloadInterface
      */
     public function launch(): void
     {
-        $ltiMessage = $this->getLtiMessageOrRedirectToLogin();
+        try {
+            $ltiMessage = $this->getValidatedLtiMessagePayload();
 
-        $user = $this->getServiceLocator()
-            ->getContainer()
-            ->get(tao_models_classes_UserService::class)
-            ->addUser(
+            $user = $this->getUserService()->addUser(
                 $ltiMessage->getUserIdentity()->getIdentifier(),
                 helpers_Random::generateString(UserService::PASSWORD_LENGTH),
                 new core_kernel_classes_Resource(
                     $this->getAuthoringRoleService()->getValidRole($ltiMessage->getRoles())
                 )
             );
 
-        $this->getServiceLocator()
-            ->getContainer()
-            ->get(LtiService::class)
-            ->startLti1p3Session($ltiMessage, $user);
+            $this->getLtiService()->startLti1p3Session($ltiMessage, $user);
+        } catch (LtiException $exception) {
+            $this->handleLtiException($exception);
+        }
 
-        $this->forward('run', null, null, $_GET);
```

Comment:
We are never redirecting to `run` anymore. Is it ok? Is that endpoint even used now?

Reply:
Maybe should we at least deprecate that method? So it will be easier to next developer as you spent quite some time to grasp it. Wdyt?

Reply:
Sure, doing that here :point_right: b323f3efa103a93b445bf3b9c5292a72d735e909

Reply:
Amazing, really apreciated! 🙌